File: Assignments/Assignment_51/program5.cpp
// Write generic program to accept N values and reverse the contents.

#include<iostream>
using namespace std;

template<class T>

void Reverse(T*arr, int iSize)
{
    int Start = 0;
    int End = iSize - 1;
    T temp;

    while(Start < End)
    {
        temp = arr[Start];
        arr[Start] = arr[End];
        arr[End] = temp;

        Start++;
        End--;
    }
}

int main()
{
    int arr[] = {10,20,30,10,30,40,10,40,10};

    for(int i = 0; i < 9; i++)
    {
        cout<<arr[i]<<"\t";   //  10 20 30 10 30 40 10 40 10
    }
    cout<<"\n";
    
    Reverse(arr,9);
    
    for(int i = 0; i < 9; i++)
    {
        cout<<arr[i]<<"\t";           // Reversed array
    }
    cout<<"\n";
}

File: Assignments/Assignment_72/program5.java
// Write a program which accept matrix from user and swap the contents of consecutive rows.

import java.util.*;

class Matrix
{
    public int iRow;
    public int iCol;
    public int Arr[][];

    public Matrix(int a, int b)
    {
        this.iRow = a;
        this.iCol = b;

        Arr = new int[iRow][iCol];
    }

    public void Accept()
    {
        int i = 0, j = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the elements of Matrix");

        for(i = 0; i < iRow; i++)
        {
            System.out.println("Enter the elements of Row : "+ (i+1));

            for(j = 0; j < iCol; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }

        sobj.close();
    }

    public void Display()
    {
        int i = 0, j = 0;

        System.out.println("Elements of Matrix are : ");

        for(i = 0; i < iRow; i++)
        {
            for(j = 0; j < iCol; j++)
            {
                System.out.print(Arr[i][j] + "\t");
            }
            System.out.println();
        }
    }

    public void SwapRows()
    {
        int i = 0, j = 0;
        int temp;

        for(i = 0 ; i < iRow-1 ; i = i + 2)
        {
            for(j = 0 ; j < iCol ; j++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i+1][j];
                Arr[i+1][j] = temp;
            }
        }
    }
}

class program5
{
    public static void main(String A[])
    {
        Matrix mobj = new Matrix(4,4);

        mobj.Accept();

        System.out.println("Before Swap");
        mobj.Display();

        mobj.SwapRows();

        System.out.println("After Swap");
        mobj.Display();
    }
}